test/libsolidity/semanticTests/functionCall/external_call_value.sol

```diff
@@ -0,0 +1,17 @@
+pragma solidity >= 0.6.0;
+
+contract C {
+    function g(uint n) external payable returns (uint, uint) {
+        return (msg.value * 1000, n);
+    }
+
+    function f(uint n) public returns (uint, uint) {
+        return this.g{value: 10}(n);
+    }
+}
+
+// ====
+// compileViaYul: also
+// ----
+// g(uint256), 1 ether: 4 -> 1000000000000000000000, 4
+// f(uint256), 1 ether: 2 -> FAILURE
```

Comment:
Doesn't `isoltest` support constructors? Could add a `payable` constructor here so this would pass as well.

Reply:
oh, or just increase the ether amount

Reply:
Actually no, this shouldn't even fail!

Reply:
Why shouldn't it out of gas?

Reply:
Actually not out of gas

Reply:
@chriseth did you figure why this fails?

Reply:
Did not yet have the time.

Reply:
``f`` has to be ``payable``?

Reply:
oh wow. this took a while...